### ISSUE TITLE
Theme does not work with rST Vscode Extension

### ISSUE THREAD

**Issue by twodrops** [jbms/sphinx-immaterial#163]
I really like the immaterial theme, but it does not work with [rST VsCode Extension](https://marketplace.visualstudio.com/items?itemName=lextudio.restructuredtext). The problem is with the language server ([Esbonio](https://github.com/swyddfa/esbonio)) used by the extension that it crashes with the async Google font download.

I did not find a way to turn off the font download. It would be nice, if the font download is made optional.

**Comment by 2bndy5** [comment]
Can you post anything that tells us why the font is causing the extension to crash?

If it is a problem with the path to the fonts not being relative to the sphinx-build output, then you can use the config option [`sphinx_immaterial_external_resource_cache_dir`](https://jbms.github.io/sphinx-immaterial/customization.html#confval-sphinx_immaterial_external_resource_cache_dir) to tell the theme to download the fonts to your project's docs' build folder.

**Comment by twodrops** [comment]
Unfortunately the extension does not seem to write much about this problem on the console even with debug log level.
> /usr/local/lib/python3.10/site-packages/esbonio/lsp/sphinx/__init__.py:164: RuntimeWarning: coroutine 'add_google_fonts.<locals>.do_fetch' was never awaited

[sphinx_immaterial_external_resource_cache_dir](https://jbms.github.io/sphinx-immaterial/customization.html#confval-sphinx_immaterial_external_resource_cache_dir) worked well with local build but not with the build triggered by the extension,

There is some suggestion by the Esbonio developers here
https://github.com/swyddfa/esbonio/issues/451#issuecomment-1255633191

**Comment by twodrops** [comment]
Maybe it is notable to say that I am using a VsCode DevContainer. But most other themes work fine with this setup and Esbonio.

**Comment by twodrops** [comment]
@2bndy5 I found another issue which is potentially due to nesting of asyncio.

I have a python module which is documented with autodoc. The module has a "async def" method and immaterial theme does not seem to like this. Other themes work fine with the module.

```python
async def run(
    name: str,
    start_commands: Sequence[Dict[str, Any]] = ()
) -> None:
    """Runs something.

    :param name: a unique name.
    :param start_commands: some command.
    """
```

Throws during sphinx-build

> module.py:docstring of module.run:1: WARNING: Parameter name 'start_commands' does not match any of the parameters defined in the signature: ['name', 'start_commands=()']

As far as I know, autodoc loads the python modules to be rendered using python when sphinx is running. Could it be that the "async def" used in Immaterial Theme causes this to be nested and fails?

**Comment by 2bndy5** [comment]
Solution released in [v0.11.4](https://pypi.org/project/sphinx-immaterial/0.11.4/)
